src/WebJobs.Extensions/Extensions/Timers/Scheduling/CronSchedule.cs

```diff
@@ -32,13 +61,133 @@ internal CrontabSchedule InnerSchedule
+        /// so that they are unambiguous, depending on the time of schedule it is (interval or point-in-time).
+        /// Uses TimeZoneInfo.Local to make the determination.
+        /// </summary>
+        /// <param name="now">The DateTimeOffset that represents the current time.</param>
+        /// <param name="next">The DateTimeOffset that represents the next occurrence.</param>
+        /// <param name="adjusted">The adjusted DateTimeOffset, if "next" is invalid.</param>
+        /// <returns>True if the 'next' time was adjusted. Otherwise, false.</returns>        
+        internal bool TryAdjustAmbiguousTime(DateTimeOffset now, DateTimeOffset next, out DateTimeOffset? adjusted)
+        {
+            adjusted = null;
+
+            TimeZoneInfo timeZone = TimeZoneInfo.Local;
+
+            // Begin evaluating scenarios when Daylight Saving ends and time falls back. This leads to "ambiguous" times
+            // as the clock repeats an hour. For example, times from 1:00 - 1:59 AM will occur twice in Pacific Standard Time
+            // and are therefore considered "ambiguous" for this time zone.            
+            bool isNowAmbiguous = timeZone.IsAmbiguousTime(now);
+            bool isNextAmbiguous = timeZone.IsAmbiguousTime(next);
+
+            if (!isNowAmbiguous && !isNextAmbiguous)
+            {
+                // There are no ambiguous times to adjust.
+                return false;
+            }
+
+            // We also need to differentiate between "interval" and "point-in-time" schedules when exiting Daylight Saving Time.
+            //
+            // For "interval" schedules, we want to continue running through this ambiguous hour as usual. Using an "every 30 minute" schedule
+            // and Pacific time offset an example, we'd want to:
+            //    - Run at 01:00-7, 01:30-7, 01:00-8, 01:30-8, 02:00-8, 02:30-8, etc.
+            //
+            // For "point-in-time" schedules, we only want to run them once (i.e a 1:30 trigger only runs once on this day). Using an
+            // "every day at 01:30" schedule and Pacific time offset as an example in 2018, we'd want to:
+            //    - Run on November 3rd at 01:30-7
+            //    - Run on November 4th at 01:30-7 (and not run at 1:30-8)
+            //    - Run on November 5th at 01:30-8
+            if (!IsInterval && isNextAmbiguous && next.Offset != now.Offset)
+            {
+                // "Fall Back" Scenario 1 -- Point-in-time schedule where the next time is ambiguous and the offsets have changed.
+                //   This means that "now" is in DST and "next" is in Standard Time. Since we never want to run an ambiguous
+                //   point-in-time schedule on Standard time, move the offset back to the Daylight offset.
+                var offsetDiff = next.Offset - now.Offset;
+                adjusted = TimeZoneInfo.ConvertTime(next.Add(offsetDiff), timeZone);
```

Comment:
I'm having a hard time following this offset adjustment logic. Can you tie it to your example above, or update the example above to explicitly give examples for these two different scenarios?